HOMEWORK_28/EX1/EX1.cpp

```diff
@@ -0,0 +1,96 @@
+#include <iostream>
+#include <vector>
+#include <thread>
+#include <future>
+#include <algorithm>
+#include <numeric>
+#include <chrono>
+
+// ������������ ��������� QuickSort
+void quickSortSingleThread(std::vector<int>& vec, int left, int right) {
+    if (left >= right) return;
+
+    int pivot = vec[right];
+    int partitionIndex = left;
+
+    for (int i = left; i < right; ++i) {
+        if (vec[i] <= pivot) {
+            std::swap(vec[i], vec[partitionIndex]);
+            ++partitionIndex;
+        }
+    }
+    std::swap(vec[partitionIndex], vec[right]);
+
+    quickSortSingleThread(vec, left, partitionIndex - 1);
+    quickSortSingleThread(vec, partitionIndex + 1, right);
+}
+
+// �������������� ��������� QuickSort
+void quickSortMultiThread(std::vector<int>& vec, int left, int right, int depth) {
+    if (left >= right) return;
+
+    int pivot = vec[right];
+    int partitionIndex = left;
+
+    for (int i = left; i < right; ++i) {
+        if (vec[i] <= pivot) {
+            std::swap(vec[i], vec[partitionIndex]);
+            ++partitionIndex;
+        }
+    }
+    std::swap(vec[partitionIndex], vec[right]);
+
+    if (depth > 0) {
+        auto future = std::async(std::launch::async, quickSortMultiThread, std::ref(vec), left, partitionIndex - 1, depth - 1);
+        quickSortMultiThread(vec, partitionIndex + 1, right, depth - 1);
+        future.get();
```

Comment:
strange implementation... why paralleling only one part, not both?
and also why mixing up threading flow and actual partitioning? Should be in distinct functions